From 29d469eb268695b2a85f6f4f906177c2e040f98e Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 13 Jul 2020 05:18:42 +0000
Subject: [PATCH 2/2] opee: Apply a new CFG_TZDRAM_START calcutation logics

1) Allocate at the uppere edge of the 1G
2) Set the maximum available DRAM_SIZE=4G

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 core/arch/arm/plat-imx/conf.mk           | 8 +++++++-
 core/arch/arm/plat-imx/platform_config.h | 6 +++---
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index 5acb46c7..6442bb84 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -329,7 +329,13 @@ CFG_UART_BASE ?= UART1_BASE
 endif
 
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mmevk))
-CFG_DDR_SIZE ?= 0x80000000
+CFG_DDR_SIZE = 0x100000000
+CFG_DDR_0SEC_SIZE = 0x40000000
+CFG_TZDRAM_START = ($(CFG_DRAM_BASE) - 0x02000000 + $(CFG_DDR_0SEC_SIZE))
+
+CFG_DRAM0_NSEC_SIZE = ($(CFG_DDR_0SEC_SIZE) - 0x02000000 )
+CFG_DRAM1_NSEC_BASE = 0x80000000
+CFG_DRAM1_NSEC_SIZE = ($(CFG_DDR_SIZE) - $(CFG_DDR_0SEC_SIZE))
 CFG_UART_BASE = UART3_BASE
 endif
 
diff --git a/core/arch/arm/plat-imx/platform_config.h b/core/arch/arm/plat-imx/platform_config.h
index 7032237d..ac857368 100644
--- a/core/arch/arm/plat-imx/platform_config.h
+++ b/core/arch/arm/plat-imx/platform_config.h
@@ -74,9 +74,9 @@
  */
 #if defined(CFG_MX8MQ) || defined(CFG_MX8MM) || defined(CFG_MX8MN)
 #define DRAM0_NSEC_BASE CFG_DRAM_BASE
-#define DRAM0_NSEC_SIZE (CFG_TZDRAM_START - CFG_DRAM_BASE)
-#define DRAM1_NSEC_BASE (CFG_TZDRAM_START + 0x2000000)
-#define DRAM1_NSEC_SIZE ((CFG_DRAM_BASE - DRAM1_NSEC_BASE) + CFG_DDR_SIZE)
+#define DRAM0_NSEC_SIZE CFG_DRAM0_NSEC_SIZE
+#define DRAM1_NSEC_BASE CFG_DRAM1_NSEC_BASE
+#define DRAM1_NSEC_SIZE CFG_DRAM1_NSEC_SIZE
 #endif
 
 #if defined(CFG_IMX_DEK_HAB) && !defined(DRAM0_NSEC_SIZE)
-- 
2.17.1

